rustls/src/common_state.rs

```diff
@@ -25,6 +26,202 @@ use alloc::vec::Vec;
 
 use pki_types::CertificateDer;
 
+#[allow(dead_code)]
+pub(crate) enum CommonState2<'a> {
```

Comment:
`CommonState2` definitely needs a new name. Maybe it should not exist at all and we could pass around a `CommonState` and an `Option<CommonBuffers>`?

Reply:
My interpretation of this is it's a incremental migration to a new replacement type? ie

- add a new type with a temporary name
- replace instances of an old type with the new one incrementally, in successive commits
- delete the old type now it is unused
- rename the new type to the old name

Once you're done, all that can get squashed together and the temporary name disappears.